components/calendly_v2/calendly_v2.app.mjs

```diff
@@ -1,11 +1,278 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "calendly_v2",
-  propDefinitions: {},
+  propDefinitions: {
+    user: {
+      type: "string",
+      label: "User UUID",
+      description: "An user UUID",
+      async options({ prevContext }) {
+        const organizations = (await this.getUserOrganizations())
+          .collection
+          .map((e) => e.organization);
+
+        let users = [];
+        for (const organization of organizations) {
+          prevContext.organization = organization;
+          const members = (await this._makeAsyncOptionsRequest({
+            prevContext,
+            requestType: "listOrganizationMembers",
+            optionsCallbackFn: this._getUserOptions,
+          })).options;
+
+          users = [
+            ...users,
+            ...members,
+          ];
+        }
+
+        return users;
+      },
+    },
```

Comment:
I agree that the user organizations probably don't need to be paginated. I think this will only retrieve the first page of users for each organization though. My suggestion would be to add an `organization` prop with async options. then you can pass in the organization to the `user` prop.
```
    organization: {
      type: "string",
      label: "Organization",
      description: "The organization UUID",
      async options() {
        const prevContext = await this.defaultUser();
        return await this._makeAsyncOptionsRequest({
          prevContext,
          requestType: "getUserOrganizations",
          optionsCallbackFn: this._getOrganizationOptions,
        });
      },
    },
    user: {
      type: "string",
      label: "User UUID",
      description: "An user UUID",
      async options({
        prevContext, organization,
      }) {
        prevContext.organization = organization;
        return await this._makeAsyncOptionsRequest({
          prevContext,
          requestType: "listOrganizationMembers",
          optionsCallbackFn: this._getUserOptions,
        });
      },
    },
```
```
    _getOrganizationOptions(collection) {
      return {
        options: collection.map((e) => ({
          label: e.organization.split("/").pop(),
          value: e.organization,
        })),
      };
    },
```

Reply:
Yeah, that makes sense. I think this approach is a lot better. Thanks for the help